tests/components/fing/__init__.py

```diff
@@ -0,0 +1,25 @@
+"""Tests for the Fing integration."""
+
+from unittest.mock import patch
+
+from homeassistant.components.fing.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def init_integration(
+    hass: HomeAssistant, mocked_entry, mocked_fing_agent
+) -> MockConfigEntry:
+    """Set up the Mocked Fing integration."""
+
+    entry = MockConfigEntry(domain=DOMAIN, data=mocked_entry)
+    with patch(
+        "homeassistant.components.fing.coordinator.FingAgent",
+        return_value=mocked_fing_agent,
+    ):
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
```

Comment:
Why don't we create an fixture for this that can be used in both the integration and the config flow?

Example:

```py
@pytest.fixture
def mock_overseerr_client() -> Generator[AsyncMock]:
    """Mock an Overseerr client."""
    with (
        patch(
            "homeassistant.components.overseerr.coordinator.OverseerrClient",
            autospec=True,
        ) as mock_client,
        patch(
            "homeassistant.components.overseerr.config_flow.OverseerrClient",
            new=mock_client,
        ),
    ):
        client = mock_client.return_value
        client.get_request_count.return_value = RequestCount.from_json(
            load_fixture("request_count.json", DOMAIN)
        )
        yield client
```